drivers/flash/spi_nor.c

```diff
@@ -480,36 +483,43 @@ static int spi_nor_write(const struct device *dev, off_t addr,
 	}
 
 	acquire_device(dev);
+	ret = spi_nor_write_protection_set(dev, false);
+	if (ret == 0) {
```

Comment:
I think it would be easier to review if this didn't shift the entire 30+ line while loop over in order to stick in the conditional clause.  A `goto out;` if write protection setting failed seems reasonable.

Reply:
Personally I prefer to avoid `goto`. I did similar changes In a few places more. Do you want that I apply your comments to them all?

Reply:
You don't have to do it anywhere, it's just making things harder to review.  IMO.

Reply:
The `ret == 0` could just be taken into the `while` condition, this would reduce the change to single line, no need to indent code.